app/models/puppetclass.rb

```diff
@@ -110,8 +107,8 @@ def all_host_ids
     ids.uniq
   end
 
-  def update_total_hosts
-    update_attribute(:total_hosts, all_host_ids.size)
+  def hosts_count
+    all_host_ids.count
```

Comment:
It's going to be more then 2 queries because of the hostgroup ancestry (a host can get a puppetclass from it's hostgroup's parent's parent's config group for example), but I changed this to push what I can into the database.